src/bandersnatch_filter_plugins/prerelease_name.py

```diff
@@ -31,13 +32,13 @@ def initialize_plugin(self):
             ]
             logger.info(f"Initialized prerelease plugin with {self.patterns}")
 
-    def filter(self, metadata):
+    def filter(self, metadata: Dict) -> bool:
         """
         Remove all release versions that match any of the specified patterns.
         """
         releases = metadata["releases"]
         for version in list(releases.keys()):
-            if any(pattern.match(version) for pattern in self.patterns):
+            if any(pattern.match(version) for pattern in self.patterns):  # type: ignore
```

Comment:
Will be interested to see what the error is here.

Reply:
```
prerelease_name.py:41: error: Item "None" of "Optional[List[Pattern[str]]]" has no attribute "__iter__" (not iterable)
```

Reply:
Ahh - makes sense.

Yeah, here we just need to add another check for a None object and handle it. I’ll go back and look at type ignores before closing the issue and see if we can get rid of a few more.